components/movementsensor/adxl345/adxl345.go

```diff
@@ -41,9 +41,30 @@ var modelName = resource.NewDefaultModel("accel-adxl345")
 
 // AttrConfig is a description of how to find an ADXL345 accelerometer on the robot.
 type AttrConfig struct {
-	BoardName              string `json:"board"`
-	I2cBus                 string `json:"i2c_bus"`
-	UseAlternateI2CAddress bool   `json:"use_alternate_i2c_address,omitempty"`
+	BoardName              string              `json:"board"`
+	I2cBus                 string              `json:"i2c_bus"`
+	UseAlternateI2CAddress bool                `json:"use_alternate_i2c_address,omitempty"`
+	SingleTap              *TapAttrConfig      `json:"single_tap,omitempty"`
+	FreeFall               *FreeFallAttrConfig `json:"free_fall,omitempty"`
+}
+
+// TapAttrConfig is a description of the configs for tap registers.
+type TapAttrConfig struct {
+	AccelerometerPin int    `json:"accelerometer_pin"`
+	EchoInterrupt    string `json:"interrupt_pin"`
+	ExcludeTapX      bool   `json:"exclude_tap_x,omitempty"`
+	ExcludeTapY      bool   `json:"exclude_tap_y,omitempty"`
+	ExcludeTapZ      bool   `json:"exclude_tap_z,omitempty"`
+	ThreshTap        byte   `json:"thresh_tap,omitempty"`
```

Comment:
let's exclude all the user-facing exclude fields, and just report all of them in readings. The user can pick which is important in their client logic.

I would go for a small default duration and a large threshold tap. You can add these configs to the mapping later.